### ISSUE TITLE
Add labels app and version on deployments for compatibility with istio

### ISSUE THREAD

**Issue by msvticket** [jenkins-x/jx#2705]
Jenkins-X already has integrations with Istio and it seems like there are more to come. It would then make sense to improve compability with Istio.

Looking at https://istio.io/docs/setup/kubernetes/spec-requirements/ I find that the helm charts of the build packs differ from the Istio requirements by lacking the lables app and version.

So I propose adding these labels to deployments in the build packs.

**Comment by ccojocar** [comment]
@msvticket are you refereeing to build pods? Which kind of installation are you running classic or serverless?

**Comment by msvticket** [comment]
No, not build pods. Build packs. Taking maven as an example the labels app and version are missing from the deployment here https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes/blob/master/packs/maven/charts/templates/deployment.yaml

I'm not running serverless, but I don't think that matters.

**Comment by ccojocar** [comment]
Thanks. It think the app label is already there https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes/blob/4b98c965ddfc6f8a7fd1ffebf6d1757b6d9acc71/packs/maven/charts/templates/deployment.yaml#L14.

**Comment by msvticket** [comment]
Right. The headline in the istio document was misleading: "Deployments with app and version labels". But it really is the pods that should have the labels. I'll reword this story.

I think it would be useful to have the label app on the deployment as well, but that isn't istio related.

**Comment by tomhobson** [comment]
This is cool. Again we could put this forward to GSOC as I said in your previous issue. If there's still interest and this isn't fixed, let's do it!

**Comment by msvticket** [comment]
I still think it would be worthwhile.

It's a small fix in all the deployments defined in [jx3-pipeline-catalog](https://github.com/jenkins-x/jx3-pipeline-catalog).

**Comment by Arpit529Srivastava** [comment]
@msvticket hello! 
so you are saying to add labels to all the files that exists in this 
```
 https://github.com/jenkins-x/jx3-pipeline-catalog/tree/master/packs
``` 
am i correct ?

**Comment by msvticket** [comment]
> @msvticket hello! so you are saying to add labels to all the files that exists in this
> 
> ```
>  https://github.com/jenkins-x/jx3-pipeline-catalog/tree/master/packs
> ```
> 
> am i correct ?

Not exactly, I say add labels to pods in deployments.

**Comment by Arpit529Srivastava** [comment]
Okay I can work on the issue 
@msvticket can you assign it to me

**Comment by Arpit529Srivastava** [comment]
@msvticket so i have change each this file?
```
https://github.com/jenkins-x/jx3-pipeline-catalog/blob/master/packs/go-mongodb/charts/templates/deployment.yaml
```

**Comment by msvticket** [comment]
Yes

**Comment by Arpit529Srivastava** [comment]
added a PR ```https://github.com/jenkins-x/jx3-pipeline-catalog/pull/1579```
@msvticket Review it and let me know if you want any changes.
